src/components/locationRegistration/AddCity.vue

```diff
@@ -32,16 +34,44 @@ export default class AddCity extends Vue {
   private city: CityRequest = new CityRequest("");
   private showModal: boolean = false;
 
+  @Prop()
+  public cityId: string = "";
+
+  @Prop()
+  public title: string = "Voeg een stad toe";
+
   async addCity() {
-    cityService
-      .post(this.city)
-      .then(() => {
-        this.showModal = true;
-        this.city.Name = "";
-      })
-      .catch((err) => {
-        this.emitter.emit("err", err);
+    if (this.cityId) {
+      // Update.
+      this.city.Id = this.cityId;
+      cityService
+        .updateCity(this.city)
+        .then(() => {
+          this.city.Name = "";
+          this.$emit("location-changed");
+        })
+        .catch((err) => {
+          this.emitter.emit("err", err);
+        });
+    } else {
+      cityService
+        .post(this.city)
+        .then(() => {
+          this.showModal = true;
+          this.city.Name = "";
+        })
+        .catch((err) => {
+          this.emitter.emit("err", err);
+        });
+    }
+  }
+
+  async mounted() {
+    if (this.cityId) {
+      cityService.getById(this.cityId).then((res) => {
```

Comment:
zag dat je bij AddBuilding aan res wel een type geeft (res: Building).
Is volgens mij wel netter ook al doen we het verder nergens. kies maar of je er zin in hebt om dat hier aan te passen :P